compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs

```diff
@@ -136,6 +136,15 @@ pub enum DefIdOrName {
     Name(&'static str),
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+enum ErrorSortKey {
+    SubtypeFormat(usize, usize),
+    OtherKind,
+    ClauseTraitSized,
+    Coerce,
+    ClauseWellFormed,
+}
```

Comment:
It's not immediately obvious what this enum is for. This really needs a doc comment, something like "for sorting the error messages so that sized and WF obligations come last". But I personally would put this inside of the function, just above `errors.sort_by_key(|e| {`. That way it is obvious that it's used for that.